components/brave_extension/extension/brave_extension/background/today/feedToData.ts

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+import { formatDistanceToNow } from 'date-fns'
+import hashStrings from '../../../../../common/hashStrings'
+
+function shuffleArray(array: Array<any>) {
+  for (let i = array.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [array[i], array[j]] = [array[j], array[i]];
+  }
+}
+
+export function generateRelativeTimeFormat (publishTime: string) {
+  if (!publishTime) {
+    return
+  }
+  // TODO(petemill): proper internationalisation syntax
+  return formatDistanceToNow(
+    new Date(publishTime)
+  ) + ' ago'
+}
+
+function convertFeedItem(item: BraveToday.FeedItem): BraveToday.FeedItem {
+  const publishTime = item.publish_time + ' UTC'
+  return {
+    ...item,
+    img: item.padded_img,
+    publish_time: publishTime,
+    relative_time: generateRelativeTimeFormat(publishTime),
+  }
+}
+
+export default async function getBraveTodayData (
```

Comment:
Lots of logic in here - could be useful to break specific parts out to a new method(s) if we need to unit test specific parts. That could always be done at a later time (follow up) 😄

Reply:
Yeah I was thinking I'd unit test this whole function. Already broken out `take<T>`, `generateNextPage` and the weighting, but can also break out all the category calculations.